python/BioSimSpace/Sandpit/Exscientia/FreeEnergy/_restraint_search.py

```diff
@@ -0,0 +1,1113 @@
+                if not exes:
+                    raise _MissingSoftwareError("Cannot use AMBER engine as AMBER is not installed!")
+                elif len(exes) > 1:
+                    _warnings.warn(f"Multiple AMBER engines were found. Proceeding with {exes[0]}...")
+                self._exe = exes[0]
+
+        else:
+            # Use SOMD as a default.
+            engine = "SOMD"
+
+        # Set the engine.
+        self._engine = engine
+
+        if not isinstance(ignore_warnings, bool):
+            raise ValueError("'ignore_warnings' must be of type 'bool'.")
+        self._ignore_warnings = ignore_warnings
+
+        if not isinstance(show_errors, bool):
+            raise ValueError("'show_errors' must be of type 'bool'.")
+        self._show_errors = show_errors
+
+        # Check that the map is valid.
+        if not isinstance(property_map, dict):
+            raise TypeError("'property_map' must be of type 'dict'.")
+        self._property_map = property_map
+
+        # Create fake instance methods for 'analyse'. These
+        # pass instance data through to the staticmethod versions.
+        self.analyse = self._analyse
+
+        # Initialise the process.
+        self._initialise_process(self._system, gpu_support, **kwargs)
+
+    def start(self):
+        """Start the simulation."""
+        self._process.start()
+
+    def wait(self):
+        """Wait for the simulation to finish."""
+        self._process.wait()
+
+    def kill(self):
+        """Kill the process."""
+        self._process.kill()
+
+    def isRunning(self):
+        """Check if the process is running."""
+        self._process.isRunning()
+
+    def workDir(self):
+        """Return the working directory.
+
+           Returns
+           -------
+
+           work_dir : str
+               The path of the working directory.
+        """
+        return self._work_dir
+
+    def _analyse(self, rest_type='Boresch',
+                method='MDRestraintsGenerator',
+                append_to_lig_selection="",
+                recept_selection_str='protein and name CA C N',
+                cutoff=10, # In Angstrom
+                block='AUTO'):
+        """Analyse trajectory and select restraints which best mimic strongest
+        receptor-ligand interactions.
+
+           Parameters
+           ----------
+
+           rest_type: str
+               The type of restraints to select (currently only Boresch is available).
+               Default is 'Boresch'.
+
+           method: str
+               The method to use to derive the restraints. Currently only 'MDRestraintsGenerator'
+               is supported.
+
+           append_to_lig_selection: str
+               Appends the supplied string to the default atom selection which chooses
+               the atoms in the ligand to consider as potential anchor points. The default
+               atom selection is f'resname {ligand_resname} and not name H*'. Uses the
```

Comment:
Sorry, what is the problem here?

Reply:
Apologies, I see what you are doing here. Just looking at the commit diff it is unclear that these parts of the documentation aren't expanded from variables elsewhere. Looking at the full file makes things clear. Since it's not possible for the user to set the `ligand_resname` (it's taken directly from the decoupled molecule) perhaps it's useful to highlight what it means and that it's not an adjustable user parameter. (I guess they can change it if they call the private analyse methods directly, but it's not something that they need to do.)